### PR TITLE
Implied columns tests and messaging

### PR DESCRIPTION
**Pull request by vreuter**
TL;DR --> adds implied columns parsing context and tests and slightly improves existing functionality.

This adds some context to the messaging around implied columns parsing. It also adds tests for this new feature. Along for the ride is the removal of the restriction to `Series` on type of input data to `Sample` constructor, and ensuring that exception type reflects the access mode for a `Sample` data item. That is, dot-notation produces `AttributeError` and getitem syntax produces `KeyError` when the name of the data item requested isn't defined for the `Sample` instance in question. There's a small handful of tests here for those changes as well. Other changes are stylistic.

### CODE REVIEW

**Comment by vreuter** [comment]
Ticking off checklist tasks 1 & 2 from https://github.com/epigen/looper/pull/94
